src/main/scala/com/cloudant/clouseau/SupportedAnalyzers.scala

```diff
@@ -87,23 +90,14 @@ object SupportedAnalyzers {
     }
   }
 
-  def createAnalyzerInt(options: Any): Option[Analyzer] = options match {
-    case name: String =>
-      createAnalyzerInt(Map("name" -> name))
-    case list: List[(String, Any)] =>
-      try {
-        createAnalyzerInt(list.toMap)
-      } catch {
-        case e: ClassCastException => None
```

Comment:
this is important, I don't see it reintroduced elsewhere or made impossible to happen by other changes.

Reply:
I did try to move this into the ClouseauTypeFactory and implement it as a partial function. The `collect` would skip things which do not match the types defined in the `case` check. So we fail faster and don't even go to type casting.

```
def collectKVs(list: List[_]): List[(String, Any)] =
    list.collect { case t @ (_: String, _: Any) => t }.asInstanceOf[List[(String, Any)]]
```

One of my assumptions turned out to be wrong. Specifically the assumption about `default` field to be a string. Which means I have to